ec2utils/ec2uploadimg/lib/ec2utils/ec2uploadimg.py

```diff
@@ -367,6 +377,22 @@ def _create_target_root_volume(self):
            the image we are creating."""
         return self._create_volume('%s' % self.root_volume_size)
 
+    # ---------------------------------------------------------------------
+    def _create_temporary_private_key(self, key_name='temporary_ec2_uploadkey'):
+        if self.verbose:
+            print('Creating temporary key pair')
+        suffix = ''.join(random.choices(string.ascii_uppercase + string.digits, k=12))
+        self.ssh_key_pair_name = key_name + '-' + suffix
+        self.ssh_key_private_key_file = os.path.join('/tmp/', self.ssh_key_pair_name + '.key')
```

Comment:
We should not write the private key to tmp. We should create a tmpfile and keep it in memory or write it to the users home directory. tmpfile is preferred